cishell/src/main/java/org/cishell/cibridge/cishell/impl/CIShellCIBridgeSchedulerFacade.java

```diff
@@ -211,24 +234,20 @@ public Boolean setSchedulerRunning(Boolean running) {
         return cibridge.getSchedulerService().isRunning();
     }
 
-    // TODO Update the subscriptions implementation with listeners
     @Override
     public Publisher<Boolean> schedulerCleared() {
-        Boolean boolean1 = true;
-        List<Boolean> results = new ArrayList<>();
-        results.add(boolean1);
-        return Flowable.fromIterable(results).delay(2, TimeUnit.SECONDS);
-        // return Flowable.just(boolean1).;
+        Flowable<Boolean> publisher;
```

Comment:
There is a lot of repeated code code with these subscription functions. You should create a helper method to clean up the cruft..
```java
private Publisher<T> asPublisher<T>(ConnectableObservable<T> connectableObservable) {
    Flowable<T> publisher;
    publisher = connectableObservable.toFlowable(BackpressureStrategy.BUFFER);
    return publisher;
}

public Publisher<Boolean> schedulerCleared() {
     return asPublisher<Boolean>(schedulerClearedObservable);
}
```

or something like that.